iceoryx_hoofs/test/moduletests/test_relative_pointer.cpp

```diff
@@ -243,158 +279,162 @@ TYPED_TEST(RelativePointer_test, AssignmentOperatorTests)
 
     {
         auto offset = SHARED_MEMORY_SIZE + 1;
-        auto typedPtr = reinterpret_cast<TypeParam*>(ptr1 + offset);
+        // NOLINTJUSTIFICATION Pointer arithmetic needed for tests
+        // NOLINTNEXTLINE(cppcoreguidelines-pro-bounds-pointer-arithmetic, cppcoreguidelines-pro-type-reinterpret-cast)
+        auto* typedPtr = reinterpret_cast<TypeParam*>(ptr1 + offset);
         RelativePointer<TypeParam> rp;
         rp = typedPtr;
         EXPECT_NE(rp, nullptr);
     }
 }
 
-TYPED_TEST(RelativePointer_test, IdAndOffset)
+TYPED_TEST(RelativePointer_test, IdAndOffsetAreTranslatedToRawPointerCorrectly)
 {
     ::testing::Test::RecordProperty("TEST_ID", "9a29a074-d68d-4431-88b9-bdd26b1a41f7");
-    auto ptr = this->partitionPtr(0);
-    auto typedPtr = reinterpret_cast<TypeParam*>(ptr);
-
-    RelativePointer<TypeParam> rp1(typedPtr, 1);
-    EXPECT_EQ(rp1.registerPtr(1, typedPtr), true);
-    EXPECT_EQ(rp1.getOffset(), reinterpret_cast<std::ptrdiff_t>(ptr));
-    EXPECT_EQ(rp1.getId(), 1);
-
-    int offset = SHARED_MEMORY_SIZE / 2;
-    auto addressAtOffset = reinterpret_cast<TypeParam*>(ptr + offset);
-    RelativePointer<TypeParam> rp2(addressAtOffset, 1);
+    auto* ptr = this->partitionPtr(0U);
+    auto* typedPtr = static_cast<TypeParam*>(static_cast<void*>(ptr));
```

Comment:
please use reinterpret_cast

Reply:
`static_cast` has several advantages e.g. it can be used during compile-time, see [here](https://github.com/eclipse-iceoryx/iceoryx/pull/1521#discussion_r936463947).